modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -169,350 +226,319 @@ ADLAROMANCEStressUpdateBase::initQpStatefulProperties()
 }
 
 ADReal
-ADLAROMANCEStressUpdateBase::computeResidual(const ADReal & effective_trial_stress,
-                                             const ADReal & scalar)
+ADLAROMANCEStressUpdateBase::maximumPermissibleValue(const ADReal & effective_trial_stress) const
 {
-  if (_immobile_function)
-    _immobile_old = _immobile_function->value(_t, _q_point[_qp]);
-  else
-    _immobile_old = _immobile_dislocations_old[_qp];
+  // Make maximum allowed scalar a little bit less than the deformation that would reduce the trial
+  // stress to zero. This prevents negative trial stresses.
+  return (effective_trial_stress - 1.0e-9) / _three_shear_modulus;
```

Comment:
Maybe it would be better to multiply by a number slightly less than one (so it is independent of the magnitude of stress)

Reply:
Do you mean slightly more than 0?

Reply:
Oh, no i see, you're right....do you think it matters?

Reply:
Actually, yeah, if stress is really high, the 1e-9 will get lost in machine precision?